core/app/login/token/[token]/route.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * This route is used to accept customer login token JWTs from the
+ * [Customer Login API](https://developer.bigcommerce.com/docs/start/authentication/customer-login)
+ * and log the customers in using alternative authentication methods
+ */
+
+import { redirect } from 'next/navigation';
+import { signIn } from '~/auth';
+import { decodeJwt } from 'jose';
+import { isRedirectError } from 'next/dist/client/components/redirect-error';
+
+export async function GET(
+  request: Request,
+  { params }: { params: Promise<{ token: string }> }
+) {
+  const token = (await params).token;
+
+  // check if token is valid
+  try {
+    decodeJwt(token); // throws error if token is an invalid JWT
+
+    // decode token without checking signature to get redirect path
+    // token is not checked for validity here, so it could be expired or invalid at this point
+    // token validity is checked in the signIn function
+    const claims = decodeJwt(token);
+    const redirectTo = claims.redirect_to || '/account/orders';
+
+    // sign in with token which will check validity against BigCommerce API
+    // and redirect to redirectTo
+    await signIn('credentials', {
+      type: 'jwt',
+      jwt: token,
+        redirectTo
+    });
+  } catch (error) {
+    // https://github.com/nextauthjs/next-auth/discussions/9389
+    if (isRedirectError(error)) {
+      throw error;
+    }
```

Comment:
`import { unstable_rethrow as rethrow } from 'next`;

```suggestion
    rethrow(error)
```